core/iwasm/common/wasm_c_api.c

```diff
@@ -4721,17 +4772,20 @@ wasm_instance_new_with_args(wasm_store_t *store, const wasm_module_t *module,
 {
     char sub_error_buf[128] = { 0 };
     char error_buf[256] = { 0 };
-    bool import_count_verified = false;
     wasm_instance_t *instance = NULL;
     WASMModuleInstance *inst_rt;
     CApiFuncImport *func_import = NULL, **p_func_imports = NULL;
-    uint32 i = 0, import_count = 0, import_func_count = 0;
+    uint32 i = 0, import_func_count = 0;
     uint64 total_size;
-    bool processed = false;
+    bool build_exported = false;
 
     bh_assert(singleton_engine);
 
-    if (!module) {
+    if (!module)
+        return NULL;
+
+    if (!compare_imports(module, imports)) {
+        LOG_ERROR("need a valid list of import");
```

Comment:
Had better `snprintf(sub_error_buf, sizeof(sub_error_buf), ...)` and `goto failed` like L4797? So a trap can be created when error occurs.